lib/Controller/RoomController.php

```diff
@@ -1482,6 +1483,30 @@ protected function changeParticipantType(int $attendeeId,  bool $promote): DataR
 		return new DataResponse();
 	}
 
+	/**
+	 * @PublicPage
+	 * @RequireLoggedInModeratorParticipant
+	 *
+	 * @param int $attendeeId
+	 * @param int $state
+	 * @return DataResponse
+	 */
+	public function setParticipantPublishingAllowed(int $attendeeId, int $state): DataResponse {
```

Comment:
We could also store it against the session and basically reset it when the user leaves? This would allow to have a user in 1 window speaking but not in another. But might not be a real usecase most of the time.

Reply:
a possible use case for multiple devices/sessions: one mobile device acting as a camera, far away from the speaker, and another one closer to the speaker with audio only. or multiple camera views of the same person and audio on another one.

Reply:
> We could also store it against the session and basically reset it when the user leaves? This would allow to have a user in 1 window speaking but not in another. But might not be a real usecase most of the time.

I stored it in the attendee because needing to allow just one of the sessions to publish seemed like a corner case. Like if two different people were in a call with the same user. In general I think that if a participant is allowed to publish that participant should be allowed to publish from any session/device.

> a possible use case for multiple devices/sessions: one mobile device acting as a camera, far away from the speaker, and another one closer to the speaker with audio only. or multiple camera views of the same person and audio on another one.

Note that the publishing permissions are set by the moderators. I do not think that a moderator should be concerned about the setup of the participant; if a participant wants to speak the moderator should not need to know which specific camera view to allow.

Having said all that if you think it makes more sense to add it for specific sessions rather than attendees fine by me.

Reply:
Sure fine, I just had our company call in mind where it would make sense to reset the permissions after each call. Would of course be a pain when it is reset due to a reload or network issue resulting in a new session.